codec-http/src/test/java/io/netty/handler/codec/http/HttpClientCodecTest.java

```diff
@@ -332,6 +332,28 @@ public void testWebSocket00Response() {
         assertThat(ch.readInbound(), is(nullValue()));
     }
 
+    @Test
+    public void testWebDavResponse() {
+        byte[] data = ("HTTP/1.1 102 Processing\r\n" +
+                       "Status-URI: Status-URI:http://status.com; 404\r\n" +
+                       "\r\n" +
+                       "1234567812345678").getBytes();
+        EmbeddedChannel ch = new EmbeddedChannel(new HttpClientCodec());
+        assertTrue(ch.writeInbound(Unpooled.wrappedBuffer(data)));
+
+        HttpResponse res = ch.readInbound();
+        assertThat(res.protocolVersion(), sameInstance(HttpVersion.HTTP_1_1));
+        assertThat(res.status(), is(HttpResponseStatus.PROCESSING));
+        HttpContent content = ch.readInbound();
+        // HTTP 102 is not allowed to have content.
+        assertThat(content.content().readableBytes(), is(0));
+        content.release();
+
+        assertThat(ch.finish(), is(false));
+
+        assertThat(ch.readInbound(), is(nullValue()));
```

Comment:
this is redundant with the above assert. As if `finish()` returns `false` there will be nothing left to consume